src/qibo/abstractions/abstract_gates.py

```diff
@@ -205,6 +207,10 @@ def on_qubits(self, *q) -> "Gate":
             A :class:`qibo.abstractions.gates.Gate` object of the original gate
             type targeting the given qubits.
         """
+        # Note that q is interpreted as a map from the original qubits to the new ones, e.g.
+        # gates.CNOT(2, 3)._on_qubits(0, 1, 2, 3) is equivalent to gates.CNOT(2, 3)
+        # gates.CNOT(2, 3)._on_qubits(1, 2, 3, 0) is equivalent to gates.CNOT(3, 0)
```

Comment:
Let me suggest to convert these comments to an example in the docstring of this function.